src/flp/types/fixedpoint_l2.rs

```diff
@@ -0,0 +1,638 @@
+    square_computer: Vec<F>,
+    phantom: PhantomData<T>,
+    // range/position constants
+    range_norm_begin: usize,
+    range_norm_end: usize,
+}
+
+impl<T: Fixed, F: FieldElement> FixedPointBoundedL2VecSum<T, F> {
+    /// Return a new [`FixedPointBoundedL2VecSum`] type parameter. Each value of this type is a
+    /// fixed point vector with `entries` entries.
+    pub fn new(entries: usize) -> Result<Self, FlpError> {
+        // (I) Check that the fixed type `F` is compatible.
+
+        // We only support fixed types that encode values in [-1,1].
+        // These have a single integer bit.
+        if <T as Fixed>::INT_NBITS != 1 {
+            return Err(FlpError::Encode(format!(
+                "Expected fixed point type with one integer bit, but got {}.",
+                <T as Fixed>::INT_NBITS,
+            )));
+        }
+
+        // Compute number of bits of an entry, and check that an entry fits
+        // into the field.
+        let bits_per_entry: usize = (<T as Fixed>::INT_NBITS + <T as Fixed>::FRAC_NBITS)
+            .try_into()
+            .map_err(|_| FlpError::Encode("Could not convert u32 into usize.".to_string()))?;
+        if !F::valid_integer_bitlength(bits_per_entry) {
+            return Err(FlpError::Encode(format!(
+                "fixed point type bit length ({}) too large for field modulus",
+                bits_per_entry,
+            )));
+        }
+
+        // (II) Check that the field is large enough for the norm.
+
+        // Valid norms encoded as field integers lie in [0,2^(2*bits - 2)).
+        let bits_for_norm = 2 * bits_per_entry - 2;
+        if !F::valid_integer_bitlength(bits_for_norm) {
+            return Err(FlpError::Encode(format!(
+                "maximal norm bit length ({}) too large for field modulus",
+                bits_for_norm,
+            )));
+        }
+
+        // In order to compare the actual norm of the vector with the claimed
+        // norm, the field needs to be able to represent all numbers that can
+        // occur during the computation of the norm of any submitted vector,
+        // even if its norm is not bounded by 1. Because of our encoding, an
+        // upper bound to that value is `entries * 2^(2*bits - 2)` (see docs of
+        // compute_norm_of_entries for details). It has to fit into the field.
+        let usize_max_norm_value: usize = match entries.checked_mul(1 << bits_for_norm) {
+            Some(val) => val,
+            None => {
+                return Err(FlpError::Encode(format!(
+                    "number of entries ({}) not compatible with field size",
+                    entries,
+                )))
+            }
+        };
+        F::valid_integer_try_from(usize_max_norm_value)?;
+
+        Ok(Self {
+            bits_per_entry,
+            entries,
+            bits_for_norm,
+            range_01_checker: poly_range_check(0, 2),
+            square_computer: vec![F::zero(), F::zero(), F::one()],
```

Comment:
To help clarify that this is a polynomial, since we lack a decent type for that:
```suggestion
            // polynomial: 0 + 0 * x + 1 * x^2
            square_computer: vec![F::zero(), F::zero(), F::one()],
```